panel/src/panel/content.js

```diff
@@ -110,22 +110,32 @@ export default (panel) => {
 		 * Saves any changes
 		 */
 		async save() {
-			if (this.isProcessing === true) {
-				return true;
-			}
-
 			this.isProcessing = true;
 
+			// ensure to abort unfinished previous save request
+			// to avoid race conditions with older content
+			this.saveAbortController?.abort();
+			this.saveAbortController = new AbortController();
+
 			try {
-				await panel.post(this.api + "/changes/save", panel.view.props.content);
-			} finally {
+				await panel.post(this.api + "/changes/save", panel.view.props.content, {
+					signal: this.saveAbortController.signal
+				});
+
 				this.isProcessing = false;
 
 				// update the last modification timestamp
 				panel.view.props.lock.modified = new Date();
```

Comment:
I moved these inside the `try` part as these shouldn't happen for failed requests or aborted requests. Maybe `isProcessing = false` should also happen for failed requests but in that case, we would rather need a counter how many requests have been started and `isProcessing` is a computed prop with `request !== 0` logic.

Reply:
What happens on abort though? Could we reset isProcessing on abortion and then always have a correct state without a counter?

Reply:
Reset to what? `false`? But when a request is aborted, it means that another request was just started.

Reply:
Sorry, I missed the point here. The second request would have already set isProcessing to true and if we set it to false it would be totally wrong. Then we probably really need a counter.

Reply:
@bastianallgeier What about this?

```js
try {
	await panel.post(...);
	
	// update the last modification timestamp
	panel.view.props.lock.modified = new Date();

	this.isProcessing = false;
} catch (error) {
	if (error.name !== "AbortError") {
		this.isProcessing = false;
		throw error;
	}
}
```

Reply:
Oh, if that works, that'd be great!